src/al.jl

```diff
@@ -4,40 +4,52 @@ using Logging, SolverTools, NLPModels
 
 using JSOSolvers, Krylov
 
-"""Implementation of a augmented Lagrangian method for:
-
-  min f(x)  s.t.  c(x) = 0, l ≦ x ≦ u"""
+function al(nlp :: AbstractNLPModel; kwargs...)
+  if unconstrained(nlp) || bound_constrained(nlp)
+    return al(Val(:tron), nlp; kwargs...)
+  elseif equality_constrained(nlp)
+    return al(Val(:equ), nlp; kwargs...)
+  else # has inequalities
+    return al(Val(:ineq), nlp; kwargs...)
+  end
+end
 
-function al(nlp :: AbstractNLPModel; mu :: Real = eltype(nlp.meta.x0)(10.0),
+function al(::Val{:tron}, nlp :: AbstractNLPModel;
             max_iter :: Int = 1000, max_time :: Real = 30.0, max_eval :: Int=-1,
-            atol :: Real = 1e-7, rtol :: Real = 1e-7,
+            atol :: Real = 1e-8, rtol :: Real = 1e-8,
             subsolver_logger :: AbstractLogger=NullLogger(),
            )
+  if !(unconstrained(nlp) || bound_constrained(nlp))
+    error("al(::Val{:tron}, nlp) should only be called for unconstrained or bound-constrained problems. Use al(nlp)")
+  end
+  @warn "Problem doesn't have constraints, so tron is being called instead"
```

Comment:
instead of what? Here I would write "problem does not have general constraints; calling tron.".